Source/AwsCommonRuntimeKit/http/HttpClientConnection.swift

```diff
@@ -7,97 +7,36 @@ import Foundation
 
 public class HttpClientConnection {
     private let allocator: Allocator
-    private let rawValue: UnsafeMutablePointer<aws_http_connection>
-    
-    init(connection: UnsafeMutablePointer<aws_http_connection>, allocator: Allocator = defaultAllocator) {
+    let rawValue: UnsafeMutablePointer<aws_http_connection>
+    let manager: HttpClientConnectionManager
+
+    init(manager: HttpClientConnectionManager,
```

Comment:
connections can technically be created without a manager. Are you preventing that? If so, this is fine. If not, manager needs to be optional, and it'll have to be null checked elsewhere in here.

Reply:
is there some reason I am not thinking of in which I would need a connection without a manager? cuz if so I will add.